phpcs.xml.dist

```diff
@@ -172,4 +172,8 @@
 			</property>
 		</properties>
 	</rule>
+	<rule ref="Gutenberg.Commenting.FunctionCommentSinceTag">
+		<!-- The sniff ensures that functions have a valid @since tag but skips checking experimental packages. -->
```

Comment:
Nice!

We should also include `/pacakages/block-serialization-default-parser/.+/*\.php$`.

https://github.com/WordPress/wordpress-develop/blob/b1b97f96727f137210b942f00f85fc6449033d1d/tools/webpack/packages.js#L128-L135

There are two more special core blocks:
- `widgets/src/blocks/legacy-widget/index.php`
- `widgets/src/blocks/widget-group/index.php`

https://github.com/WordPress/wordpress-develop/blob/b1b97f96727f137210b942f00f85fc6449033d1d/tools/webpack/blocks.js#L38-L48

@youknowriad, what do we miss here? Should we discuss in a separate issue what else we could copy from Gutenberg to the core and put under more pressure to follow WP coding standards? 😄

Reply:
It is interesting that @getdave opened a discussion about this PR yesterday https://github.com/WordPress/gutenberg/discussions/59786

--- 

Yeah, so these files would be good to address next but I wonder if all the php files in Gutenberg should have these rules applied (Manual backports include a lot of copy/pasting of php files)

Reply:
Thank you for your feedback, @gziolo. Your review is greatly appreciated.
I'd prefer to implement checking files in `/pacakages/block-serialization-default-parser` and other folders in a follow-up issue as this would allow for incremental improvements while ensuring that the enforcement of @since tags in the Block Library is already in place.

Follow-up issue: https://github.com/WordPress/gutenberg/issues/59826

> Yeah, so these files would be good to address next but I wonder if all the php files in Gutenberg should have these rules applied (Manual backports include a lot of copy/pasting of php files)

If there's a decision to enable this sniff for all the `.php` files in Gutenberg, then, in my opinion, a mechanism is needed to determine how to skip checks in the files that won't be backported to Core.
Many functions/methods are unique to Gutenberg, and enabling this sniff for all `.php` files would require adding a lot of `// phpcs:ignore` statements to skip these Gutenberg-only functions/methods.

Reply:
Yes, it is definitely another issue/PR. We better introduce these improvements gradually so we can start catching issues for covered files.